server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -1635,10 +1635,19 @@ public Network createGuestNetwork(CreateNetworkCmd cmd) throws InsufficientCapac
             throwInvalidIdException("Network offering with specified id doesn't support adding multiple ip ranges", ntwkOff.getUuid(), NETWORK_OFFERING_ID);
         }
 
-        if (GuestType.Shared == ntwkOff.getGuestType() && !ntwkOff.isSpecifyVlan() && Objects.isNull(associatedNetworkId)) {
-            throw new CloudRuntimeException("Associated network must be provided when creating Shared networks when specifyVlan is false");
-        }
 
+
+        if (GuestType.Shared == ntwkOff.getGuestType()) {
+            if (!ntwkOff.isSpecifyIpRanges()) {
+                throw new CloudRuntimeException("The 'specifyipranges' parameter should be true for Shared Networks");
+            }
```

Comment:
there is already a check when create the network offering

https://github.com/apache/cloudstack/blob/b2b2218a8abf750c92a31bd89078d96b5c65159e/server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java#L6720-L6730

Reply:
ok, leaving it in in case someone hacks the DB.